usr/share/rear/backup/NETFS/default/500_make_backup.sh

```diff
@@ -304,7 +304,7 @@ backup in order to be sure to safely recover this system.
                 rc=$(cat $FAILING_BACKUP_PROG_RC_FILE)
                 Error "$prog failed with return code $rc and below output:
   ---snip---
-$(grep "^$prog: " "${TMP_DIR}/${BACKUP_PROG_ARCHIVE}.log" | sed -e 's/^/  /' | tail -n3)
+$(sed -n -e '/^tar: .*\(socket ignored\|Removing leading\)/d;/^'"$prog"':/s/^/  /p' "${TMP_DIR}/${BACKUP_PROG_ARCHIVE}.log" | tail -n3)
```

Comment:
The `sed ... /^'"$prog"':/s/^/  /p` part does not work when a `prog` in the pipe
shows error messages that are not prefixed with `$prog:`

It works for `prog=dd` like
```
# prog=dd

# dd -QQ 2>&1 | sed -n -e '/^tar: .*\(socket ignored\|Removing leading\)/d;/^'"$prog"':/s/^/  /p'
  dd: invalid option -- 'Q'

# dd if=/dev/urandom of=/ramdisk/foo 2>&1 | sed -n -e '/^tar: .*\(socket ignored\|Removing leading\)/d;/^'"$prog"':/s/^/  /p'
  dd: writing to '/ramdisk/foo': No space left on device
```

But it does not work for `prog=openssl` e.g. when something like
`BACKUP_PROG_CRYPT_OPTIONS="/usr/bin/openssl des3 -salt -k "`
is used:
```
# prog=openssl

# openssl QQ 2>&1 | sed -n -e '/^tar: .*\(socket ignored\|Removing leading\)/d;/^'"$prog"':/s/^/  /p'

no output because

# openssl QQ
Invalid command 'QQ'; type "help" for a list.

# openssl des3 -d -k 2>&1 | sed -n -e '/^tar: .*\(socket ignored\|Removing leading\)/d;/^'"$prog"':/s/^/  /p'

also no output because

# openssl des3 -d -k
des3: Option -k needs a value
des3: Use -help for summary.
```

So when the user specified non working BACKUP_PROG_CRYPT_OPTIONS
he could not see the matching openssl error messages.

Reply:
I assume the only clean way out is that each program in the pipe
redirects (or even better `tee`s) its stderr into a separated temporary file
so that we can show only the stderr messages of the program that failed.

Reply:
> I assume the only clean way out is that each program in the pipe
> redirects (or even better `tee`s) its stderr into a separated temporary file
> so that we can show only the stderr messages of the program that failed.

That's a possible improvement, especially for openssl, whicht is extremely verbose and does not add an "openssl: " prefix to its error messages:
```
# /usr/bin/openssl des3 -salt -kfile /foo.bar.baz
unable to read key from '/foo.bar.baz'
options are
-in <file>     input file
-out <file>    output file
-pass <arg>    pass phrase source
---8<---
-rc5-ecb                   -rc5-ofb                   -seed                     
-seed-cbc                  -seed-cfb                  -seed-ecb                 
-seed-ofb                  
```
A possible solution would be to use a wrapper, e.g.
```
# /usr/bin/openssl des3 -salt -kfile /foo.bar.baz 2> >(sed -e 's/^/openssl: /;q')
openssl: unable to read key from '/foo.bar.baz'
```
But that's a different problem that must be fixed in separate changes.

Reply:
Only FYI that different openssl versions show different output,
here what I on my openSUSE Leap 15.1 system:
```
# openssl version
OpenSSL 1.1.0i-fips  14 Aug 2018

# /usr/bin/openssl des3 -salt -kfile /foo.bar.baz
des3: Cannot open input file /foo.bar.baz, No such file or directory
des3: Use -help for summary.

# /usr/bin/openssl des3 -salt -kfile /foo.bar.baz 2> >(sed -e 's/^/openssl: /;q')
openssl: des3: Cannot open input file /foo.bar.baz, No such file or directory
```

Reply:
By the way:
Filtering stdout or stderr directly in the program call is in general not wanted,
cf. the section "What to do with stdin, stdout, and stderr" in
https://github.com/rear/rear/wiki/Coding-Style
that reads (excerpt):

All what could be useful for later debugging in case of issues must appear in ReaR's log file.
Usually command output should not appear on the user's terminal but in ReaR's log file.
Because stdout and stderr are redirected into ReaR's log file you should omit things
like '1>/dev/null' or '2>/dev/null' or '&>/dev/null' to have all messages in the log file
where they help to identify the root cause of a problem.
On the other hand if a program is needelssly verbose you may
use '1>/dev/null' or '2>/dev/null' or '&>/dev/null' as appropriate to avoid
that the log file is "polluted" with useless messages

For an example how to `tee` stdout and/or stderr into separated files see
https://github.com/rear/rear/blob/master/usr/share/rear/backup/BORG/default/500_make_backup.sh#L71
and see the explanatory comment for that code.
For background information see the discussion in
https://github.com/rear/rear/pull/
in particular
https://github.com/rear/rear/pull/2382#discussion_r417852393
and subsequent comments there like
https://github.com/rear/rear/pull/2382#discussion_r418018571
and
https://github.com/rear/rear/pull/2382#discussion_r419378558